### PR TITLE
Core & Internals: Porting PL/SQL procedure to abacus daemon; Fix #524

### LINKED ISSUE #524: Porting oracle jobs/procedures to non-oracle backends
Motivation
----------
At the moment there are two oracle procedures (cron jobs) being used to keep the data consistent. However, there needs to be a solution for non-oracle users. For Postgres it might be possible to also introduce PLSQL Cronjobs, but for mysql this is not possible, thus daemon implementations for this might be applicable.

- Update collection replicas: Exists only as PL/SQL
- Update of account counters: Exists as PL/SQL and in the Abacus daemon

### PR DESCRIPTION
**Pull request by hahahannes**
Core & Internals: Porting PL/SQL procedure to abacus daemon; Fix #524

### CODE REVIEW

File: lib/rucio/core/replica.py [hunk 1]
```diff
@@ -2261,3 +2263,130 @@ def mark_unlocked_replicas(rse, bytes, session=None):
...
+    update_requests_without_rse_id = []
+    for update_request in replica_update_requests:
+        if update_request.rse_id is not None:
+            small_request = {'name': update_request.name, 'scope': update_request.scope, 'rse_id': update_request.rse_id}
+            if small_request not in update_requests_with_rse_id:
+                update_requests_with_rse_id.append(small_request)
+            else:
+                update_request.delete(session=session)
+                continue
+            found_replicas = session.query(models.CollectionReplica).filter(models.CollectionReplica.rse_id == update_request.rse_id, models.CollectionReplica.scope == update_request.scope, models.CollectionReplica.name == update_request.name).all()
```
Comment (mlassnig) [inline lib/rucio/core/replica.py]: you can cut line length, just put a backslash and start the next line with the dot of the function

(happens a couple of times, you can do it everywhere)

File: lib/rucio/core/replica.py [hunk 2]
```diff
@@ -2261,3 +2263,130 @@ def mark_unlocked_replicas(rse, bytes, session=None):
...
+            if len(found_replicas) == 0:
```
Comment (mlassnig) [inline lib/rucio/core/replica.py]: shorter:
`if not found_replicas:`

Comment (mlassnig) [reply lib/rucio/core/replica.py]: (below as well)

File: lib/rucio/tests/test_abacus_account.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+
+class TestAbacusAccount():
+
+    def setUp(self):
+        self.account = 'root'
+        self.scope = 'mock'
+        self.upload_client = UploadClient()
+        self.file_sizes = 2
+        self.rse = 'MOCK4'
```
Comment (mlassnig) [inline lib/rucio/tests/test_abacus_account.py:44]: use rse_name_generator()

Comment (hahahannes) [reply lib/rucio/tests/test_abacus_account.py:44]: For this case I wanted to use one of the simulated RSEs where I can upload a real file. I am not sure if this is possible with a random RSE.

Comment (mlassnig) [reply lib/rucio/tests/test_abacus_account.py:44]: understood

**Pull request merged**